packages/ui/v2/modules/List.tsx

```diff
@@ -3,18 +3,31 @@ import { createElement } from "react";
 
 import classNames from "@calcom/lib/classNames";
 
-export function List(props: JSX.IntrinsicElements["ul"]) {
+export type ListProps = {
+  roundContainer?: boolean;
+} & JSX.IntrinsicElements["ul"];
+
+export function List(props: ListProps) {
   return (
-    <ul {...props} className={classNames("-mx-4 rounded-sm sm:mx-0 sm:overflow-hidden", props.className)}>
+    <ul
+      {...props}
+      className={classNames(
+        "-mx-4 rounded-sm sm:mx-0 sm:overflow-hidden ",
+        // Add rounded top and bottome if roundContainer is true
+        props.roundContainer && "[&>*:first-child]:rounded-t-md [&>*:last-child]:rounded-b-md ",
+        props.className
+      )}>
       {props.children}
     </ul>
   );
 }
 
-export type ListItemProps = { expanded?: boolean } & ({ href?: never } & JSX.IntrinsicElements["li"]);
+export type ListItemProps = { expanded?: boolean; rounded?: boolean } & ({
+  href?: never;
+} & JSX.IntrinsicElements["li"]);
 
 export function ListItem(props: ListItemProps) {
-  const { href, expanded, ...passThroughProps } = props;
+  const { href, expanded, rounded = true, ...passThroughProps } = props;
```

Comment:
See comment above 

If the above approached worked we would be able to get away without this additional prop